### PR TITLE
Refactor: Split azure_backoff and azure_wrap in provider package

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
No logic changes.  just split files. 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_interface_repo_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (lzhecheng) [inline pkg/provider/azure_interface_repo_test.go:2]: ditto

File: pkg/provider/azure_privatelinkservice_repo_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/*
```
Comment (lzhecheng) [inline pkg/provider/azure_privatelinkservice_repo_test.go:1]: If there's no UT originally, how about adding some or skipping this empty file?

File: pkg/provider/azure_subnet_repo_test.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*
```
Comment (lzhecheng) [inline pkg/provider/azure_subnet_repo_test.go:1]: ditto

File: pkg/provider/azure_vmsets_repo.go [hunk 4]
```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (lzhecheng) [inline pkg/provider/azure_vmsets_repo.go:2]: 2023

File: pkg/provider/azure_vmss_repo.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (lzhecheng) [inline pkg/provider/azure_vmss_repo.go:2]: ditto

File: pkg/provider/azure_vmss_repo_test.go [hunk 6]
```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (lzhecheng) [inline pkg/provider/azure_vmss_repo_test.go:2]: ditto

File: pkg/provider/azure_interface_repo.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/provider/azure_interface_repo.go]: fixed. Thanks!

File: pkg/provider/azure_vmsets_repo.go (hunk 4 above)
Comment (MartinForReal) [reply pkg/provider/azure_vmsets_repo.go]: fixed. Thanks!

File: pkg/provider/azure_vmss_repo.go (hunk 5 above)
Comment (MartinForReal) [reply pkg/provider/azure_vmss_repo.go]: fixed. Thanks!

File: pkg/provider/azure_subnet_repo_test.go (hunk 3 above)
Comment (MartinForReal) [reply pkg/provider/azure_subnet_repo_test.go:1]: fixed. Thanks!

File: pkg/provider/azure_interface_repo.go [hunk 8]
```diff
@@ -0,0 +1,39 @@
+/*
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/provider/azure_interface_repo.go:1]: exposed functions are CRUD functions, we may need to change the signature and move to repo package.

File: pkg/provider/azure_privatelinkservice_repo_test.go (hunk 2 above)
Comment (MartinForReal) [reply pkg/provider/azure_privatelinkservice_repo_test.go:1]: removed. Thanks!

**Comment by lzhecheng** [comment]
The purpose of this PR is a prerequisite of SDK track2 migration so it is helpful to have individual files for resource CRUD. In the future, CCM will switch to track2 but also reserve the possibility of switching back. Finally, these files will be removed.

**Pull request merged**